### PR TITLE
Availability controllers

### PR DESCRIPTION
**Pull request by kozaktar**
Get availability Controller:
usage:
GET /availability?day=DAY*)

day must be in iso-string format and time set to start of the day. 

Returns an array of the form:

[
{start: date-time, end: date-time}
.
.
.
{start: date-time, end: date-time}
]

### CODE REVIEW

File: server/middleware/auth.js [hunk 1]
```diff
@@ -12,8 +12,8 @@ const auth = (req, res, next) => {
             return;
         } else {
             const decodedToken = jwt(jwtToken);
-            const { userId } = decodedToken;
-            req.userId = userId;
+            const { user_id } = decodedToken;
```
Comment (alvyjudy) [inline server/middleware/auth.js]: I think we agreed during the standup yesterday to use camel case for userId. You might need to change in the `authentication.js` how the token is generated

File: server/routes/authentication.js [hunk 2]
```diff
@@ -17,99 +17,81 @@ const oAuthClient = new google.auth.OAuth2(
...
+    if (app_user.length > 0) {
+      claims = { user_id: app_user[0]._id, email };
+    } else {
+      const newUser = new User({
+        email: userInfo.payload.email,
+        name: userInfo.payload.name
+      });
+      try {
+        await newUser.save();
+        claims = { user_id: newUser._id, email };
```
Comment (alvyjudy) [inline server/routes/authentication.js]: should be `claims = {userId: newUser._id, email}`

File: server/db/models/AuthenticationStore.js [hunk 3]
```diff
@@ -3,26 +3,26 @@ const mongoose = require("mongoose");
...
+            "Please fill a valid email address",
+        ],
+    },
+    //token used for api access and google auth
+    googleAuthToken: {
+        type: Schema.Types.Mixed,
+    },
+    refreshToken: {
+        type: Schema.Types.Mixed,
+    },
```
Comment (alvyjudy) [inline server/db/models/AuthenticationStore.js:25]: indent spacing

File: server/db/models/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
-  Appointment: require("./Appointment.js"),
-  EventType: require("./EventType.js"),
-  User: require("./User"),
+    Appointment: require("./Appointment.js"),
+    EventType: require("./EventType.js"),
+    User: require("./User"),
+    AuthStore: require("./AuthenticationStore")
```
Comment (alvyjudy) [inline server/db/models/index.js:5]: indent spacing

File: server/routes/googleCalendarAPI.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    catch(err){
+        res.status(500).send("error getting calendar availability")
+    }
+   
+
+});
+
+
+
+module.exports = router;
```
Comment (alvyjudy) [inline server/routes/googleCalendarAPI.js:29]: indent spacing

File: server/utils/credentials.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"client_id":"632445597219-hlqdt77sscv5s66o1apmscsgkn2nd3qr.apps.googleusercontent.com","project_id":"quickstart-1611166970191","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"tpn5NL89VKsXUTtw-uX2mGv5","redirect_uris":["http://localhost:3000/authorizedd"]}
```
Comment (alvyjudy) [inline server/utils/credentials.json:1]: shouldn't we keep these in the .env file?

**Review by alvyjudy — CHANGES_REQUESTED** [review]

File: server/utils/credentials.json (hunk 6 above)
Comment (kozaktar) [reply server/utils/credentials.json:1]: that was only for testing. I'll delete it

**Comment by alvyjudy** [comment]
It looks like you have reverted a lot of changes from the dev branch when you merged. Looks like my comments were for the old code. They should all be resolved after you sync your branch with the dev correctly!

**Pull request merged**